remotefile/src/qrmf_socketadapter.cpp

```diff
@@ -9,7 +9,22 @@
 #include "qnumheader.h"
 #include "qscan.h"
 
-#define RMF_SOCKET_BUFFER_SIZE_LIMIT 32 * 1024
+#ifndef RMF_MAX_EXPECTED_MESSAGE_LENGTH
+#define RMF_MAX_EXPECTED_MESSAGE_LENGTH 1024 * 1024
+#endif
+
+#ifndef RMF_SOCKET_BUFFER_SIZE_LIMIT
+# if RMF_MAX_EXPECTED_MESSAGE_LENGTH >= (64 * 1024)
+#      define RMF_SOCKET_BUFFER_SIZE_LIMIT (RMF_MAX_EXPECTED_MESSAGE_LENGTH / 2)
+#   else
+#      define RMF_SOCKET_BUFFER_SIZE_LIMIT 32 * 1024
+#   endif
+#endif
+
+// Define to get additional debug info on large transactions
+//#define RMF_SOCKET_MESSAGE_LENGTH_DEBUG
+#define RMF_SOCKET_MESSAGE_LENGTH_DEBUG_DUMP_BYTES 24

```

Comment:
I would have rearranged these lines and gather the options close together so the developer gets an easier understanding of what options can be enable/disabled during build.

#define RMF_SOCKET_MESSAGE_LENGTH_DEBUG_DUMP_BYTES 24

/*** Options ***/
// Define to get additional debug info on large transactions
+//#define RMF_SOCKET_MESSAGE_LENGTH_DEBUG

// Define to get qDebug socket status
//#define RMF_SOCKET_VERBOSE